### ISSUE TITLE
CodeFormatter should accept SourceText

### ISSUE THREAD

**Issue by belav** [belav/csharpier#621]
There are cases where someone has a `SourceText` and being forced to convert it to a `string` is an extra step that shouldn't be required. `CodeFormatter` could pretty easily accept `SourceText` as well.

**Comment by TwentyFourMinutes** [comment]
I actually noticed that a `SourceText` is already a [finalized visual form](https://sourceroslyn.io/#Microsoft.CodeAnalysis/Text/SourceText.cs,095fb020b503d1a4) of the `SyntaxTree`, so it therefore really should instead have an overload accepting the actual `SyntaxTree`, what do you think?

**Comment by belav** [comment]
Yeah that makes sense. Maybe it has an overload for both of them?

**Comment by belav** [comment]
`CSharpSyntaxTree.ParseText` has an overload that accepts a `SourceText` that you may have missed.

**Comment by TwentyFourMinutes** [comment]
Oh I think that we are both talking about different things. When I am creating a source code through the `SyntaxFactory` API I am currently first converting the `SyntaxTree` > `SourceText` > `string`. Now from my perspective it would make the most sense if I could directly pass the `CodeFormatter` the `SyntaxTree`. Not because it is necessarily less code to write, but rather because it is a lot faster, as you are converting the passed `string`/`SourceText` back to a `SyntaxTree` as far as I am aware of.

**Comment by TwentyFourMinutes** [comment]
Oh I see 😄. Well that would make sense however do you think that it should be encourage to pass it a `SourceText`? `CSharpSyntaxTree.ParseText` will more or less convert it to a string anyway and the performance hit would just be hidden. Also I can't think of any quick scenario in my head where I'd have the `SourceText`, but not the SyntaxTree representation of a piece of code.

**Comment by belav** [comment]
I am good with just having the overload for `SyntaxTree`, no need to add another overload if no one is going to need it.
